### PR TITLE
Add scanpy

### PR DESCRIPTION
**Pull request by Sanaz01**

### CODE REVIEW

File: scvi-tools/Dockerfile_old [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+RUN conda config --set always_copy true
+
+# Installing necessary libraries via conda
+RUN conda install anaconda::pandas=2.0.3
+RUN conda install conda-forge::matplotlib=3.8.4
+RUN conda install conda-forge::scanpy=1.10.1
+RUN conda install conda-forge::leidenalg=0.10.2
+RUN conda install pytorch torchvision torchaudio pytorch-cuda=11.7 -c pytorch -c nvidia
+RUN conda install jax jaxlib -c conda-forge
+RUN pip install scvi-tools==1.1.2
```
Comment (tefirman) [inline scvi-tools/Dockerfile_old:21]: Same thing here, are all these packages necessary for scvi-tools to run properly? The idea of this library is to make containers as decoupled tool-wise as possible.

**Review by tefirman — CHANGES_REQUESTED** [review]

File: scanpy/Dockerfile
[the remark this answers is not in the document]
Comment (Sanaz01) [reply scanpy/Dockerfile:1]: Rightly pointed out. I have made suggested changes to make it light weight and swapped conda with pip installation.

File: scvi-tools/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+LABEL org.opencontainers.image.description="Container image for the use of scvi-tools library for Single-Cell analysis in Python"
+LABEL org.opencontainers.image.version="latest"
+LABEL org.opencontainers.image.authors="wilds@fredhutch.org"
+LABEL org.opencontainers.image.url=https://hutchdatascience.org/
+LABEL org.opencontainers.image.documentation=https://getwilds.org/
+LABEL org.opencontainers.image.source=https://github.com/getwilds/wilds-docker-library
+LABEL org.opencontainers.image.licenses=MIT
+
+# Installing necessary libraries via pip
+RUN pip install pandas==2.2.2 matplotlib==3.8.4 scanpy==1.10.1 leidenalg==0.10.2 torch==2.2.2 torchvision==0.17.2 \
```
[the remark this answers is not in the document]
Comment (Sanaz01) [reply scvi-tools/Dockerfile]: PIp installs all necessary packages, so it should not be needed. Removing them from commits.

File: scvi-tools/Dockerfile_old (hunk 1 above)
Comment (Sanaz01) [reply scvi-tools/Dockerfile_old]: Removing double installation of packages. Pytorch carries good weight as it provides GPU capabilities.

File: scanpy/Dockerfile
Comment (tefirman) [inline scanpy/Dockerfile:1]: For the GitHub action to properly build and push these images, you need to add at tag at the end of the Dockerfile name separated by an underscore, i.e. "Dockerfile_{tag}". Mind renaming these to "Dockerfile_latest"?

File: scvi-tools/Dockerfile
Comment (tefirman) [inline scvi-tools/Dockerfile:1]: For the GitHub action to properly build and push these images, you need to add at tag at the end of the Dockerfile name separated by an underscore, i.e. "Dockerfile_{tag}". Mind renaming these to "Dockerfile_latest"?

**Review by tefirman — CHANGES_REQUESTED** [review]
First off, holy crap, really sorry it took me so long to review... 🤦‍♂️ Those updates to pip look awesome, thanks for updating those. One last thing to update though, just for WILDS-style purposes, mind adding the tag to the filenames? (see comments for details).

**Comment by Sanaz01** [comment]
No worries at all. I have changed the Dockerfile names and updated package version as well!

**Review by tefirman — CHANGES_REQUESTED** [review]
Interesting update! Yesterday, I noticed that the linter GitHub Action [wasn't performing its check](https://github.com/getwilds/wilds-docker-library/actions/runs/10534862581), turns out we've recently added some security features in WILDS to restrict which action types are allowed. Just needed to add the `hadolint` action to the list of acceptable actions and it ran as expected on the next push.

That being said, you'll now notice that [the check failed](https://github.com/getwilds/wilds-docker-library/actions/runs/10539686976/job/29203521080?pr=25) because of a few small optimization issues with `scvi-tools` (follow link or see screenshot below). Mind fixing these as well? Happy to chat Monday to go through these as well!

image

**Review by tefirman — APPROVED** [review]
Found one last issue. I tried building both images locally and the scvi-tools image errored out. Looks like combining the two `pip` calls threw things off because of the `--index-url` argument. I just split it into two `pip` calls again, but on the same `RUN` command, compiled fine from there. Thanks for your hard work and patience on this Sanaz! 🥳

**Pull request merged**